get_http.go

```diff
@@ -220,6 +262,44 @@ func (g *HttpGetter) parseMeta(r io.Reader) (string, error) {
 	}
 }
 
+func GetByteRange(u *url.URL) ([]string, error) {
```

Comment:
1. I think this can be unexported `getByteRange` so we don't have to commit to public API stability of this, and you may want to consider attaching it to `(*HttpGetter)`.

2. I'd also document this method (even unexported). It'd help to note the precondition that `u` is non-nil and is also not modified. It'd also help to note the response permutations: `(nil, nil)`, `(nil, non-nil)`, `(non-nil, non-nil)`, `(non-nil, nil)`. Not all may be possible but usually my primary question with APIs in Go. In this case I think the pertinent ones are `(nil, nil) - no byte range specified, no error`, `(nil, non-nil) - byte range maybe specified, error`, `(non-nil, nil) - byte range with len == 2, idx0 < idx1`.

Note the quality of #2 is highly inconsistent across many Go codebases, but when I review I try to help champion some high quality comments. :)

Reply:
Document where?

Reply:
Above, as a standard godoc-ish comment above the function.

The note I made was that as an unexported function godoc wouldn't show it of course but I still find that documenting methods is an important practice.